elements/lisk-bft/src/finality_manager.ts

```diff
@@ -365,42 +361,8 @@ export class FinalityManager extends EventEmitter {
 			return true;
 		}
 
-		// Order the two block headers such that earlierBlock must be forged first
-		let earlierBlock = validatorLastBlock;
-		let laterBlock = blockHeader;
-		const higherMaxHeightPreviouslyForged =
-			earlierBlock.asset.maxHeightPreviouslyForged > laterBlock.asset.maxHeightPreviouslyForged;
-		const sameMaxHeightPreviouslyForged =
-			earlierBlock.asset.maxHeightPreviouslyForged === laterBlock.asset.maxHeightPreviouslyForged;
-		const higherMaxHeightPrevoted =
-			earlierBlock.asset.maxHeightPrevoted > laterBlock.asset.maxHeightPrevoted;
-		const sameMaxHeightPrevoted =
-			earlierBlock.asset.maxHeightPrevoted === laterBlock.asset.maxHeightPrevoted;
-		const higherHeight = earlierBlock.height > laterBlock.height;
-		if (
-			higherMaxHeightPreviouslyForged ||
-			(sameMaxHeightPreviouslyForged && higherMaxHeightPrevoted) ||
-			(sameMaxHeightPreviouslyForged && sameMaxHeightPrevoted && higherHeight)
-		) {
-			[earlierBlock, laterBlock] = [laterBlock, earlierBlock];
-		}
-
-		if (
-			earlierBlock.asset.maxHeightPrevoted === laterBlock.asset.maxHeightPrevoted &&
-			earlierBlock.height >= laterBlock.height
-		) {
-			/* Violation of the fork choice rule as validator moved to different chain
-			 without strictly larger maxHeightPreviouslyForged or larger height as
-			 justification. This in particular happens, if a validator is double forging. */
-			throw new BFTForkChoiceRuleError();
-		}
-
-		if (earlierBlock.height > laterBlock.asset.maxHeightPreviouslyForged) {
-			throw new BFTChainDisjointError();
-		}
-
-		if (earlierBlock.asset.maxHeightPrevoted > laterBlock.asset.maxHeightPrevoted) {
-			throw new BFTLowerChainBranchError();
+		if (areHeadersContradicting(validatorLastBlock, blockHeader)) {
+			throw new BFTError();
```

Comment:
Seems good, just one concern here. Having particular errors e.g. `BFTChainDisjointError`, `BFTForkChoiceRuleError` were useful in debugging. If we can still keep those and throw those particular errors from within `areHeadersContradicting` that would nice.

Reply:
this function will be used in both contradicting and non contradicting case, so it would be nice to have boolean check IMO.
I cannot think of a way to keep the boolean return value and not throwing error in a nice way here